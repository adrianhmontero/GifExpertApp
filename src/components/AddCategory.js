import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories, categories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    if (categories.every((cat) => cat !== inputValue) && inputValue)
      setCategories((prev) => [inputValue, ...prev]);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
};
