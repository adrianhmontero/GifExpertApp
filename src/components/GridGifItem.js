import PropTypes from "prop-types";
import React from "react";

export const GridGifItem = ({ title, url }) => {
  return (
    <div className="container-gif animate__animated animate__fadeInRight">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GridGifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
