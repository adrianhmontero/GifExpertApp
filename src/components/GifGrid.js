import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GridGifItem } from "./GridGifItem";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  if (loading) return <p className="animate__animated animate__flash">Cargando gifs...</p>;

  return (
    <>
      <h3 className="animate__animated animate__fadeIn" key={category}>
        {category}
      </h3>
      <div className="gifs-grid">
        {images.map((gif) => (
          <GridGifItem key={gif?.id} {...gif} />
        ))}
      </div>
    </>
  );
};
