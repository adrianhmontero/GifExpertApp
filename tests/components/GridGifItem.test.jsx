import { render, screen } from "@testing-library/react";
import { GridGifItem } from "../../src/components/GridGifItem";

describe("Pruebas en <GridGifItem />", () => {
  const title = "Luffy",
    url = "https://one-punch.com/saitama.jpg";

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<GridGifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe mostrar el titulo en el componente", () => {
    render(<GridGifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
