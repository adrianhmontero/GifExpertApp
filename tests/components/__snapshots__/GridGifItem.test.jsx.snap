// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridGifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="container-gif animate__animated animate__fadeInRight"
  >
    <img
      alt="Luffy"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Luffy
    </p>
  </div>
</div>
`;
